src/validation.cpp

```diff
@@ -3768,6 +3768,7 @@ bool Chainstate::InvalidateBlock(BlockValidationState& state, CBlockIndex* pinde
                 // Do not remove candidate from the highpow_outofchain_headers cache, because it might be a descendant of the block being invalidated
                 // which needs to be marked failed later.
             }
+            // Best header updating currently doesn't use CBlockIndexWorkComparator, so nChainWork is used for consistency.
```

Comment:
In commit "doc: Improve doc for candidate blocks" (e32e72039f02ea0b565e0ef5ab7fa40faeb65e4c)

Would maybe s/nChainWork/only nChainWork/ since using nChainWork shouldn't be surprising here, but only using nChainWork might be.